src/Appwrite/Platform/Workers/Databases.php

```diff
@@ -563,22 +563,19 @@ protected function deleteByGroup(string $collectionId, array $queries, Database
         $start = \microtime(true);
 
         try {
-            $documents = $database->deleteDocuments($collectionId, $queries);
+            $count = $database->deleteDocuments(
+                $collectionId,
+                $queries,
+                Database::DELETE_BATCH_SIZE,
+                $callback
+            );
         } catch (\Throwable $th) {
             $tenant = $database->getSharedTables() ? 'Tenant:'.$database->getTenant() : '';
             Console::error("Failed to delete documents for collection:{$database->getNamespace()}_{$collectionId} {$tenant} :{$th->getMessage()}");
             return;
         }
 
-        if (\is_callable($callback)) {
-            foreach ($documents as $document) {
-                $callback($document);
-            }
-        }
-
         $end = \microtime(true);
-        $count = \count($documents);
-
         Console::info("Deleted {$count} documents by group in " . ($end - $start) . " seconds");
```

Comment:
as we have return inside catch block we never reach this block if catch is called, so this can safely be moved inside try block?

Reply:
I think it's better to keep the try/catch scope as small as possible - the timing/logging code isn't exceptional